addons/web/static/src/core/utils/autoresize.js

```diff
@@ -0,0 +1,74 @@
+/** @odoo-module **/
+
+import { useRefListener } from "./hooks";
+import { useEffect } from "@odoo/owl";
+
+/**
+ * This is used on text inputs or textareas to automatically resize it based on its
+ * content each time it is updated. It takes the reference of the element as
+ * parameter and some options. Do note that it may introduce mild performance issues
+ * since it will force a reflow of the layout each time the element is updated.
+ *
+ * @param {Ref} ref
+ */
+export function useAutoresize(ref, options = {}) {
+    useEffect(() => {
+        if (ref.el) {
+            resize();
+        }
+    });
+    useRefListener(ref, "input", resize);
+    function resize() {
+        if (ref.el instanceof HTMLInputElement) {
+            autoResizeInput(ref.el);
+        } else if (ref.el instanceof HTMLTextAreaElement) {
+            autoResizeTextArea(ref.el, options);
+        }
+    }
+}
+
+function autoResizeInput(input) {
+    // This mesures the maximum width of the input which can get from the flex layout.
+    input.style.width = "100%";
+    const maxWidth = input.clientWidth;
+    // Minimum width of the input
+    input.style.width = "10px";
```

Comment:
not sure I understand this. can't you remove it? and if not, move it below the 2 ifs?

Reply:
This is used to prevent some x-axis overflow when trying to expand the width of the input too much. It is mandatory to compute it before resetting the width of the input afterwards if we want this to have the right effect